osbenchmark/worker_coordinator/runner.py

```diff
@@ -967,6 +968,97 @@ async def _scroll_query(opensearch, params):
                 "took": took
             }
 
+        async def _vector_search_query_with_recall(opensearch, params):
+            """
+            Perform vector search and report recall@k , recall@r and time taken to perform recall in ms as
+            meta object.
+            """
+            result = {
+                "weight": 1,
+                "unit": "ops",
+                "success": True,
+                "recall@k": 0,
+                "recall@1": 0,
+            }
+
+            def _is_empty_search_results(content):
+                if content is None:
+                    return True
+                if "hits" not in content:
+                    return True
+                if "hits" not in content["hits"]:
+                    return True
+                if len(content['hits']['hits']) == 0:
+                    return True
+                return False
+
+            def calculate_recall(predictions, neighbors, top_k):
+                """
+                Calculates the recall by comparing top_k neighbors with predictions.
+                recall = Sum of matched neighbors from predictions / total number of neighbors from ground truth
+                Args:
+                    predictions: list containing ids of results returned by OpenSearch.
+                    neighbors: list containing ids of the actual neighbors
+                    neighbors for a set of queries
+                    top_k: number of top results to check from the neighbors and should be greater than zero
+                Returns:
+                    Recall between predictions and top k neighbors from ground truth
+                """
+                correct = 0.0
+                if neighbors is None:
+                    self.logger.info("No neighbors are provided for recall calculation")
+                    return 0.0
+                min_num_of_results = min(top_k, len(neighbors))
+                truth_set = neighbors[:min_num_of_results]
+                for j in range(min_num_of_results):
+                    if j >= len(predictions):
+                        self.logger.info("No more neighbors in prediction to compare against ground truth.\n"
```

Comment:
Pylint raises a warning about this line since you're using fstrings in logging statements. Please conform and use lazy % interpolation. 
```
self.logger.info("No more neighbors in prediction to compare against ground truth.\n"
    "Total neighbors in prediction: %s. \n"
    "Total neighbors in ground truth: %s", len(predictions), min_num_of_results)
```

Reply:
Ack